### PR TITLE
feat(calendar): add slots

### LINKED ISSUE #2393: 日历加入插槽
- [ ] I'm sure this does not appear in [the issue list of the repository](https://github.com/arco-design/arco-design-vue/issues)

## What are the similar cases of this feature
麻烦尽快更新一下，大佬，现在用的其它的插件，效果不太好，也有点丑，麻烦尽快更新，谢谢大佬

## What problem does this feature solve?
现在只是展示作用吧，我如果想做考勤或者待办事项等功能没法用吧

### LINKED ISSUE #2658: calendar组件文档没有自定义单元格插槽
- [ ] I'm sure this does not appear in [the issue list of the repository](https://github.com/arco-design/arco-design-vue/issues)

## What are the similar cases of this feature
目前现在没有这个功能，和文档

## What problem does this feature solve?
目前现在没有这个功能，和文档

### PR DESCRIPTION
**Pull request by kinggq**
## Types of changes

- [x] New feature

## Background and context

This PR introduces two new slots to the Calendar component:

 - A header slot for custom header content.
 - A default slot for rendering custom content in each date cell.

The addition of these slots increases the flexibility and customization options for the Calendar component.
## Solution

Added two slots to the Calendar component:
 - header slot: Allows users to customize the header section of the calendar.
 - Default slot: Allows users to provide custom content to be rendered in each date cell.
 - These slots are optional and do not break existing functionality.

## How is the change tested?

Manually tested to ensure that the slots work as expected.
## Changelog

| Component | Changelog(CN) | Changelog(EN) | Related issues |
| --------- | ------------- | ------------- | -------------- |
|     Calendar      |     新增 `header` 和 `default` 插槽        |       Added `header` and `default` slots        |   Close #2658 , Close #2393             |

## Checklist:

- [x] Test suite passes (`npm run test`)
- [x] Provide changelog for relevant changes (e.g. bug fixes and new features) if applicable.
- [x] Changes are submitted to the appropriate branch (e.g. features should be submitted to `feature` branch and others

## Other information

This feature enhances the Calendar component without introducing any breaking changes. Users can opt to use these slots for more advanced customization, or continue to use the component as-is without any changes.

### CODE REVIEW

**Comment by kaikaibenkai** [comment]
close #2393

**Comment by kinggq** [comment]
> @kinggq ++ close #2393 

What?

**Comment by kaikaibenkai** [comment]
```diff
- Related issues: #2658
+ Related issues: #2658 #2393
```

File: packages/web-vue/components/calendar/month.tsx [hunk 1]
```diff
@@ -159,6 +160,11 @@ export default defineComponent({
                     <div class={`${prefixCls}-date-circle`}>{col.date}</div>
                   )}
                 </div>
+                {slots.default?.({
```
Comment (flsion) [inline packages/web-vue/components/calendar/month.tsx:163]: 日期的 slot 插槽可以放到 154 行，将 calendar-date 元素包裹起来。允许用户完全自定义里面的内容

**Pull request merged**